src/breadboard/breadboard-connections.cpp

```diff
@@ -401,31 +364,26 @@ void Breadboard::removeDevice(const DeviceID& id) {
 }
 
 void Breadboard::clear() {
-    vector<gpio::PinNumber> iofs;
     for(auto const& [id,spi] : m_spi_channels) {
-        iofs.push_back(spi.gpio_offs);
+        m_embedded->closeIOF(spi.global_pin);
     }
     for(auto const& [id,pin] : m_pin_channels) {
-        iofs.push_back(pin.gpio_offs);
+        m_embedded->closeIOF(pin.global_pin);
     }
-    emit(clearIOFs(iofs));
-
-    setMinimumSize(DEFAULT_SIZE);
-    setBackground(DEFAULT_PATH);
-}
-
-void Breadboard::clearObjects() {
     m_spi_channels.clear();
     m_pin_channels.clear();
     m_writing_connections.clear();
     m_reading_connections.clear();
     m_raster.clear();
     m_devices.clear();
+
+    setMinimumSize(DEFAULT_SIZE);
+    setBackground(DEFAULT_PATH);
 }
 
 /* Update */
 
-void Breadboard::timerUpdate(gpio::State state) {
+void Breadboard::timerUpdate(uint64_t state) {
```

Comment:
wenn möglich, den type nicht "hard" einstellen, sondern an einer Stelle mit `typedef uint64_t Aarons_special_state` definieren.
Das macht den Sinn von `state` offensichtlicher, und lässt die Option offen, es später einfacher zu vergrößern/-kleinern.
Außerdem lässt das dann entspannt `static_assert()`s zu, um z.b. zur compilezeit zu checken, ob erforderliche Zahlen in den Typen passen.